### DISCUSSION TITLE
How to install OpenMage using DDEV in Windows?

### DISCUSSION THREAD

**Discussion by ankushx** [OpenMage/magento-lts#3876] [Questions & Answers]
how to install magento 1 (mageone) on ddev .give me simple steps please i have been trying for 2 days not able to login in admin page

**Comment by addison74** [comment]
This tutorial is pinned in the Discussions section.

**Test Environment for OpenMage in Windows 10 Based on DDEV**
https://github.com/OpenMage/magento-lts/discussions/3839

**Reply by ankushx** [reply]
i have install magento 1 in ddev  but my admin is not getting logged in , can you please tell me which zip file to download mageone and how to setup database with that, i am trying for 2 day in xamp as well as ddev

**Reply by addison74** [reply]
Check this resource related for using DDEV https://github.com/OpenMage/magento-lts/blob/main/docs/DDEV.md.

```
2. By default, running the ddev config command does not create an administrator account.
If you want to create or update one run in the terminal window ddev openmage-admin and follow the steps.
```

**Reply by addison74** [reply]
Forget about WAMP/XAMPP because they are slow. Don't use DDEV directly in Windows based on Mutagen. Follow my tutorial for setting up a test environment in Windows, based on WSL2, Docker, Ubuntu/Debian. This simulates a production environment almost perfect.

**Reply by ankushx** [reply]
its not working ddev openmage-admin 
![image](https://github.com/OpenMage/magento-lts/assets/114287684/53907eac-e965-4c00-af93-4475d1e8748c)

**Reply by ankushx** [reply]
i m using docker only

please help me to install mageone, 
this step i follow right now
- unzip magento in ddev
- ddev config
- ddev start
- ddev get ddev/ddev-phpmyadmin
- ddev restart
- remove app/etc/local.xml 
- go to site to set up database
- also i tried creating admin user using sql command thats also not wokring

**Reply by addison74** [reply]
1. It is good you run OpenMage in WSL2 + Ubuntu.
2. If you follow my tutorial you will find out I was using the ```git clone``` command, to clone the OpenMage repository.
3. The command ```ddev openmage-admin``` is not running because you don't have the actions files in the ```.ddev``` directory. Check if you have in your file system these two directories from here https://github.com/OpenMage/magento-lts/tree/main/.ddev.

If you have the environment set up correctly it takes around 2 - 3 minutes to download and start OpenMage.

**Reply by ankushx** [reply]
can you please send me the complete setup instruction , i have tried using git clone as well

**Reply by addison74** [reply]
Please read carefully my posts, I already pointed you where to find the tutorial for setting up the test environment.

```https://github.com/OpenMage/magento-lts/discussions/3839```

**Reply by ankushx** [reply]
its too difficult to understand. it explain more how to install ddev on windows, i already have this
i only want to install mageone

gh repo clone OpenMage/magento-lts.....this to clone right??

**Reply by ankushx** [reply]
now after cloning your repositories i m stuck here plz help

- git clone https://github.com/OpenMage/magento-lts.git
-  ddev config
- ddev start
- ddev get ddev/ddev-phpmyadmin
- ddev restart
- remove app/etc/local.xml
i m getting this 
![image](https://github.com/OpenMage/magento-lts/assets/114287684/76bade48-c14f-470b-9317-61f47066cb7a)

**Reply by fballiano** [reply]
to install openmage there's a guide on the website https://www.openmage.org/magento-lts/install.html

regarding your problem, I think with ddev you can't use localhost, but also you forgot to run composer install,

**Reply by addison74** [reply]
Let's read again the documentation from here 

https://github.com/OpenMage/magento-lts/blob/main/docs/DDEV.md#installing-openmage-in-the-browser

**Reply by ankushx** [reply]
thanks you so much its work now ,

**Reply by addison74** [reply]
All you have to do is reading the two resources carefully. I gathered there all knowledge baggage to set up a professional test environment for OpenMage. 

There was no need to install OpenMage in the browser. The following commands were enough:

```
ddev config => config the project
ddev composer install => install dependencies, you can use --no-dev too.
ddev start => start the project, in the background the database is setting up
ddev openmage-install => install Magento Sample Data
ddev openmage-admin => create or update the admin account
```

You have to enable the cron to be able to check periodic tasks such as sending messages, alerts, ... in Mailpit.

**DDEV is really a fantastic tool!**

**Reply by ankushx** [reply]
how can i install a extension in magento mageone? where can i see in admin dashboard?

**Reply by addison74** [reply]
You cannot install it in the Backend. An extension can be installed by copying the files to the root directory or using ```composer```. It depends on how the extension/module is provided by the developer. Don't forget to disable all the cache types in development.

**Reply by ankushx** [reply]
like in app/code we install a module in magento 2?  is that not possible?

**Reply by addison74** [reply]
The way OpenMage works is different than Magento 2. Let's take this extension as an example
https://github.com/fballiano/openmage-image-cleaner

I am using two variants for installing an extension/module:

**Variant 1**: Download the archive from the repository then extract/copy the ```app``` directory in the root directory of OpenMage

**Variant 2**: Run the command ```ddev composer require fballiano/openmage-image-cleaner``` in the terminal window.

**Comment by kiatng** [comment]
For a short time, I am not sure if @ankushx is talking about OpenMage (who we are) or [Mage One](https://mage-one.com/) (the other Magento 1 thing). If you are talking about OpenMage and want to use a shorter representation, use OM; avoid "mageone" or "magento 1" to not cause confusion.

**Reply by ankushx** [reply]
actually i dont know the difference in all three of them

**Reply by addison74** [reply]
You may search on the Internet as well.

**Magento 1**: EOL since 2020. Last version 1.9.4.5. Archives still here https://pubfiles.nexcess.net/magento/ce-packages/. It was replaced by Magento Commerce 2 / Magento Open Source 2.

**Mage One**: Paid services for Magento 1. Details here https://mage-one.com/.

**OpenMage**: continues Magento 1. It is the project where you asked the question.

**Reply by ankushx** [reply]
wow i didnt know about this , thnx for the usefull information

**Reply by addison74** [reply]
If you want to learn about Magento 1 / OpenMage just access this resource https://github.com/fballiano/awesome-openmage. It is a great collection.
